### ISSUE TITLE
Request cannot be constructed from a URL that includes credentials

### ISSUE THREAD

**Issue by roddone** [isomorphic-git/isomorphic-git#678]
Hi, 

I'm using isomorphic-git in the browser and the upgrade from 0.43.1 to 0.51.0 broke our connexion to azure devops git with the following error : 
TypeError: Failed to execute 'fetch' on 'Window': Request cannot be constructed from a URL that includes credentials
The url is formatted like this : https://<user>@dev.azure.com/<Organization>/<Project>/_git/<Repository>
(I can't give the exact url for privacy)

Maybe caused by #605 ?

**Comment by wmhilton** [comment]
Wait a minute... can Azure DevOps git can be fetched in the browser without using a CORS proxy?

**Comment by roddone** [comment]
Yes, it seems to be able to fetch without cors proxy 😉. 

Thanks for the quick fix, I will test it asap and close this if it corrects the issue.

**Comment by wmhilton** [comment]
@roddone It looks like Azure needs the `noGitSuffix: true` param. Are you using that? (I really want to fix that behavior for the `1.0` release.)

Also, it looks like you can't "mix" using the URL for the username and specifying the password via a parameter. E.g.
```js
git.clone({ ..., url: 'https://isomorphic-git@dev.azure.com/isomorphic-git/isomorphic-git/_git/isomorphic-git', password: 'TOKEN' })
```

throws an error saying the username is missing. I don't _think_ that worked in 0.43.1, let me know if we need to cover that case. It works if you add a redundant `username` param, though. (Since we strip out username:password from URLs now before handing them to `fetch`.)

**Comment by roddone** [comment]
@wmhilton yes we use the `noGitSuffix: true` and we specify both 'password' and 'username' parameters.
I don't think there is another way to do make it work with azure devops but if you find one, please share the tip !

I'm away from office until tomorrow, so I will probably be able to test your fix tomorrow ;)

**Comment by roddone** [comment]
I confirm your fix did the job, thanks !

**Comment by pdittaro** [comment]
Just as a headsup @roddone , you can authenticate to Azure DevOps using PATs as well if you use the `username` and `token` params. The `username` param just has to be set to the organization name.
